cli/src/semgrep/core_runner.py

```diff
@@ -835,45 +836,82 @@ def _run_rules_direct_to_semgrep_core_helper(
                 else:
                     raise SemgrepError("deep mode needs a single target (root) dir")
 
-                deep_path = SemgrepCore.deep_path()
-                if deep_path is None:
-                    raise SemgrepError(
-                        "Could not run deep analysis: DeepSemgrep not installed. Run `semgrep install-deep-semgrep`"
+                deep_path = (
+                    SemgrepCore.deep_path()
+                )  # DEPRECATED: To be removed by Feb 2023 launch
+                pro_path = SemgrepCore.pro_path()
+
+                if pro_path is None:
+                    if deep_path is None:
+                        raise SemgrepError(
+                            "Could not run deep analysis: Semgrep PRO not installed. Run `semgrep install-semgrep-pro`"
+                        )
+                    else:
+                        logger.warning(
+                            f"""You have an old DeepSemgrep binary installed in {deep_path}
+DeepSemgrep is now Semgrep PRO, run `semgrep install-semgrep-pro` to install it,
+then please delete {deep_path} manually.
+This time, we will continue running your old DeepSemgrep binary.
+"""
+                        )
+
+                if pro_path is not None:
+                    logger.info(f"Using Semgrep PRO installed in {pro_path}")
+                    version = sub_check_output(
+                        [pro_path, "-pro_version"],
+                        timeout=10,
+                        encoding="utf-8",
+                        stderr=subprocess.STDOUT,
+                    ).rstrip()
+                    logger.info(f"Semgrep PRO Version Info: ({version})")
+
+                    cmd_bin = pro_path
+                    cmd_args += [
+                        "-json_time"
+                    ]  # TODO: When do we stop adding this by default?
+                    cmd_args += ["-deep_inter_file"]
+                    cmd_args += [root]
+                elif (
```

Comment:
Couldn't we just include this section in the else case on line 849 and then turn the `if` statement on 844 to an `else`? If not, take that to mean that these conditionals are hard to understand :D

Reply:
We could, I think. I'm not sure it would be cleaner (if I'm understanding you correctly). But I didn't put much thought into cleaning this up because the idea is to remove all that `deep-semgrep` related code in the next (or next next) release..